### PR TITLE
Project Authentication - Review

### PR DESCRIPTION
**Pull request by mgunduz1**
In this project, we were implementing an authentication system using 'Devise.' So users can only access areas of the site they are authorized to access.

### CODE REVIEW

File: app/views/posts/_form.html.erb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    <div id="error_explanation">
+      <h2><%= pluralize(post.errors.count, "error") %> prohibited this post from being saved:</h2>
+
+      <ul>
+        <% post.errors.each do |error| %>
+          <li><%= error.full_message %></li>
+        <% end %>
+      </ul>
+    </div>
+  <% end %>
```
Comment (oloomoses) [inline app/views/posts/_form.html.erb]: - [ ] Kindly move this logic to helpers

File: app/views/shared/_navbar.html.erb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+         <%= link_to 'Logout', destroy_user_session_path, method: :delete, class:"nav-link active" %>
+        </li>
+        <% else %>
+        <li class="nav-item">
+          <%= link_to 'Register', new_user_registration_path, class:"nav-link active" %>
+        </li>
+        <li class="nav-item">
+          <%= link_to 'Login', new_user_session_path, class:"nav-link active" %>
+        </li>
+        <% end %>
```
Comment (oloomoses) [inline app/views/shared/_navbar.html.erb]: - [ ] Kindly move this logic to helpers

**Review by oloomoses — CHANGES_REQUESTED** [review]
## Changes required :recycle: 
**Hello team, you have done a great job on this project :muscle:** 
### Highlight
- Well done on  the linter and rubocop configuration :clap: 
- Well done on the readme and pull request. They are descriptive, simple, and easy to follow :100:
### To improve
- [x] Kindly add the .vscode file in the .gitignore. files that are not part of the project should be added in the .gitignore file
- [ ] You have logic in the views and it is very important that you move them to helpers. views should strictly be used to render content from the corresponding controller and not to generate one.
- [x] Kindly use the current user to create a new post in the post controller new action. eg `@post = current_user.posts.build(post_params)`

### Optional
- [x] It would be better to add validations to the post model to prevent the creation of an empty post

Kindly implement the above changes before submitting another review.

**Feel free to reach me on slack or leave a comment while mentioning me and I will be happy to reply**

**Review by umairarshadbutt — CHANGES_REQUESTED** [review]
# Status: REQUIRED CHANGES :warning: :construction: 

Hi @MarkoNS1990 and @mgunduz1 
Such a remarkable project you made :+1: ! Awesome README :100: !
Good job with the changes! :tada: We are almost there.
We, however, do have a few changes to carry out before we proceed.

##

Kindly work on the suggested changes  👇.
- [x] please remove `.vscode` from your repo.
- [ ] Please remove the logic from view as the previous coder reviewer suggested.
- Check this article on [Where Do I Put My Code?](http://codefol.io/posts/Where-Do-I-Put-My-Code) and [The Beginner's Guide to Rails Helpers](https://mixandgo.com/learn/the-beginners-guide-to-rails-helpers) for more info about this. Thank you. [Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/no_logic_in_helpers_requirement.png) from Pathwright.

##
- **When you have made the specified changes, please submit another code review. Thank you!**
##
**If you are stuck at any point or have any questions, let me know by leaving a comment here and mentioning me. I will be happy to help**
HAPPY CODING :computer:
**[Umair Arshad (Code Reviewer)](https://umairarshad.me/)**
[Twitter](https://twitter.com/its_UmairArshad) | [LinkedIn](https://www.linkedin.com/in/umair-arshad-butt/) | [GitHub](https://github.com/umairarshadbutt)

**Comment by mimipeshy** [comment]
## STATUS: APPROVED 🥳 👯‍♂️ 

Hello, team @mgunduz1 and @MarkoNS1990 👋🏾 👋🏾 this is such great progress you have made in this milestone. You have worked on all TSE comments 👏🏾 Therefore this project is approved, go ahead and merge. All the best in the next milestone

Regards
Peris Ndanu

**Pull request merged**